imports/plugins/included/shipping-shippo/server/methods/shippo.js

```diff
@@ -350,6 +350,65 @@ export const methods = {
     return updatingResult;
   },
 
+  /**
+   * Formats and sends a cart address to Shippo to for validation
+   * @param {String} cartId - The id of the cart to get the address from
+   * @return {Object} result - The validation result from Shippo
+   */
+  "shippo/validateCartAddress"(cartId) {
+    check(cartId, String);
+
+    const cart = Cart.findOne(cartId);
+
+    // check if cart exists and if it belongs to the current user
+    if (!cart || cart.userId !== this.userId) {
+      const msg = "Error. Your cart is either undefined or has the wrong userId.";
+      Logger.error(msg);
+      throw new Meteor.Error("address-validation-error", msg);
+    }
+
+    const apiKey = getApiKey(cart.shopId);
+
+    if (!apiKey) {
+      const msg = "No Shippo API key was found.";
+      Logger.error(msg);
+      throw new Meteor.Error("shippo-api-error", msg);
+    }
+
+    // get the buyer's email
+    const buyer = Accounts.findOne({ _id: cart.userId }, { fields: { emails: 1 } });
+
+    // check that there is address available in cart
+    if (!cart.shipping || !cart.shipping[0] || !cart.shipping[0].address) {
```

Comment:
This check won't work in a marketplace context. Generally we need to never use `shipping[0]` and eliminate wherever we see it